src/Promitor.Core.Scraping/Configuration/Serialization/v1/Core/ScrapingDeserializer.cs

```diff
@@ -1,25 +1,26 @@
 ﻿using Microsoft.Extensions.Logging;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Model;
 using YamlDotNet.RepresentationModel;
 
 namespace Promitor.Core.Scraping.Configuration.Serialization.v1.Core
 {
-    internal class ScrapingDeserializer : Deserializer<Model.ScrapingV1>
+    public class ScrapingDeserializer : Deserializer<ScrapingV1>
     {
-        internal ScrapingDeserializer(ILogger logger) : base(logger)
+        private const string ScheduleTag = "schedule";
+
+        public ScrapingDeserializer(ILogger logger) : base(logger)
         {
         }
 
-        internal override Model.ScrapingV1 Deserialize(YamlMappingNode node)
+        public override ScrapingV1 Deserialize(YamlMappingNode node)
         {
-            var scraping = new Model.ScrapingV1();
+            var scraping = new ScrapingV1();
 
-            if (node.Children.ContainsKey("schedule"))
-            {
-                var rawScheduleNode = node.Children[new YamlScalarNode("schedule")];
-                scraping.Schedule = rawScheduleNode.ToString();
-            }
-            else
+            scraping.Schedule = GetString(node, ScheduleTag);
+
+            if (scraping.Schedule == null)
             {
+                // TODO: this will log an error if this deserializer is reused to deserialize the scraping settings for a metric instead of the global scraping settings. Need to fix that!
```

Comment:
Yeah - have you got any suggestions here? It's a bit tricky because I can't alter the method signature to indicate whether the field is optional or not because it's implementing the `IDeserializer` interface. I guess I could create a separate interface for this deserializer?

Reply:
That's a good point - I'd suggest a seperate interface indeed *or* add a constructor parameter that indicates this?

Reply:
So I've had another think about this, and I'm not convinced it would be a good idea to add an option for this behaviour because it feels a bit wrong to have a parameter that alters whether a log message is output or not.

We'd end up with something like this:

```csharp
var scraping = _scrapingDeserializer.Deserialize(node, isScheduleRequired);
```

If I was using that API, I think I'd expect a parameter like that to cause an exception to be thrown, or to cause some kind of validation failure, but instead all it would do is log an error. I'm happy to just make the change if you want, but I just wanted to run it past you first in case you've got any other suggestions.

Reply:
I don't really like that as well, but maybe the ctor does make more sense? Interesting issue 🤔 

Would it make sense to create an issue for this to not block this PR?

Reply:
If you're happy with that then that's fine with me. Should we remove the message in the meantime or change it to a warning or something like that?

Reply:
Feel free to create an issue for it!

I'd suggest to change the message by adding a link to the issue.

Reply:
Actually feel free to remove it as I have a GitHub check on it